Upgrade to Java 17 and JUnit 6 (#847)
junit-pioneer/junit-pioneer: README.adoc

++++
{empty} +

> A melting pot for all kinds of extensions to https://github.com/junit-team/junit-framework[JUnit Framework] and its API.

Reviewer: Almost all of Pioneer's extensions are Jupiter extensions, right?

```suggestion
A melting pot for all kinds of extensions to https://github.com/junit-team/junit-framework[JUnit Framework] and its Jupiter API.
```

Check out https://junit-pioneer.org/[junit-pioneer.org], particularly https://junit-pioneer.org/docs/[the documentation section].

== A Pioneer's Mission

> JUnit Pioneer provides extensions for the https://github.com/junit-team/junit-framework[JUnit Framework] and its API.

Reviewer: ```suggestion
JUnit Pioneer provides extensions for the https://github.com/junit-team/junit-framework[JUnit Framework] and its Jupiter API.
```

It does not limit itself to proven ideas with wide application but is purposely open to experimentation.
It aims to spin off successful and cohesive portions into sibling projects or back into the JUnit 5 code base.

> To enable easy exchange of code with JUnit 6, JUnit Pioneer copies most of its infrastructure, from code style to build tool and configuration to continuous integration.

Reviewer: ```suggestion
To enable easy exchange of code with the JUnit Framework, JUnit Pioneer copies most of its infrastructure, from code style to build tool and configuration to continuous integration.
```

[#dependencies]
== Dependencies

Starting with release 3.0, JUnit Pioneer is compiled against **Java 17** and comes as a module (i.e. with a `module-info.class`) named _org.junitpioneer_.
That means it can be used on all Java versions 17 and higher on class path and module path.

> Pioneer does not only use JUnit 6's API, but also other artifacts from its ecosystem such as https://mvnrepository.com/artifact/org.junit.platform/junit-platform-commons[`junit-platform-commons`].

Reviewer: ```suggestion
Pioneer does not only use the JUnit Jupiter API, but also other artifacts from its ecosystem such as https://mvnrepository.com/artifact/org.junit.platform/junit-platform-commons[`junit-platform-commons`].
```

> To avoid dependency issues (e.g. in https://github.com/junit-pioneer/junit-pioneer/issues/343[junit-pioneer#343]), you should add the JUnit 6 BOM (https://mvnrepository.com/artifact/org.junit/junit-bom[`junit-bom`]) to your project instead of defining all dependency versions manually.

Reviewer: ```suggestion
To avoid dependency issues (e.g. in https://github.com/junit-pioneer/junit-pioneer/issues/343[junit-pioneer#343]), you should add the JUnit BOM (https://mvnrepository.com/artifact/org.junit/junit-bom[`junit-bom`]) to your project instead of defining all dependency versions manually.
```

> To not add to user's https://nipafx.dev/jar-hell/[JAR hell], JUnit Pioneer is not taking on any runtime dependencies besides JUnit 6.

Reviewer: ```suggestion
To not add to user's https://nipafx.dev/jar-hell/[JAR hell], JUnit Pioneer is not taking on any runtime dependencies besides JUnit.
```

> Pioneer always depends on the lowest JUnit 6 version that supports its feature set, but that should not keep you from using 6's latest and greatest.

Reviewer: ```suggestion
Pioneer always depends on the lowest JUnit Framework version that supports its feature set, but that should not keep you from using JUnit's latest and greatest.
```
